### PR TITLE
Better handling of custom RGW daemon names

### PR DESCRIPTION
**Pull request by trhoden**
If a custom RGW daemon name is given, prefix it with 'rgw.'.

Update docs and help to clarify this is done.

Fixes: http://tracker.ceph.com/issues/11561

### CODE REVIEW

**Comment by liewegas** [comment]
LGTM

File: ceph_deploy/rgw.py [hunk 1]
```diff
@@ -176,9 +176,10 @@ def rgw(args):
 
 def colon_separated(s):
     host = s
-    name = 'rgw.' + s
```
Comment (alfredodeza) [inline ceph_deploy/rgw.py]: this needs a test (or two) methinks

**Comment by trhoden** [comment]
@ktdreyer Yeah, I've been told both.  I've been told not to clutter commit messages with links to bugs, and to put them in the PRs instead.  But yeah, I agree.  Unfortunately it does not seem the link between redmine and commit messages is active for ceph-deploy.  If I put "Fixes: <id>" in the commit, it doesn't get pulled into the bug.

File: ceph_deploy/rgw.py (hunk 1 above)
Comment (trhoden) [inline ceph_deploy/rgw.py]: keeping me honest!

**Comment by trhoden** [comment]
@alfredodeza - Tests added!
@ktdreyer - commit messaged updated

**Comment by ktdreyer** [comment]
I'm curious who was advocating for not cluttering commit messages with links to bugs? We should definitely be linking to bugs in commits - it makes cherry-picking and downstream development much saner.

**Pull request merged**